src/lib/eliom_common.server.ml

```diff
@@ -1365,3 +1365,36 @@ let get_secure secure sitedata =
   match secure with
   | None -> sitedata.secure_cookies
   | Some s -> s
+
+module To_and_of_shared = struct
+
+  (* FIXME : work-around for weak polymorphism in create :( *)
+  type wrapper
+
+  type 'a t = {
+    server  : 'a to_and_of ;
+    client  : 'a to_and_of Eliom_client_value.t option;
+    wrapper : wrapper
+  }
+
+  let wrapper : wrapper = Obj.magic @@
+    Eliom_wrap.create_wrapper @@ function
+    | {client = Some tao} ->
+      tao
+    | {client = None} ->
+      failwith
+        "Cannot wrap to_and_from (missing client values for user_type?)"
```

Comment:
I think you should find a way to make that error more precise. For example, it should probably give the name of the eliom parameter.